interfaces/core.go

```diff
@@ -157,6 +157,18 @@ type Interface interface {
 	AutoConnect(plug *Plug, slot *Slot) bool
 }
 
+// Recorder describes interactions between backends and interfaces.
```

Comment:
I'm having some trouble with the terminology elsewhere. Thinking for a while, the issue is that we're lacking proper nouns for the data we're conveying. In other words, what are we recording? Why?

I'll try to provide a strawman below.. let's see if it makes sense.

// A Recorder holds the definition of plugs and slots